src/main/java/com/limechain/grandpa/GrandpaService.java

```diff
@@ -0,0 +1,246 @@
+package com.limechain.grandpa;
+
+import com.limechain.exception.global.ExecutionFailedException;
+import com.limechain.exception.storage.BlockStorageGenericException;
+import com.limechain.grandpa.state.GrandpaState;
+import com.limechain.grandpa.state.Subround;
+import com.limechain.network.protocol.grandpa.messages.commit.Vote;
+import com.limechain.network.protocol.warp.dto.BlockHeader;
+import com.limechain.storage.block.BlockState;
+import io.emeraldpay.polkaj.types.Hash256;
+import io.libp2p.core.crypto.PubKey;
+import lombok.extern.java.Log;
+import org.springframework.stereotype.Component;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Log
+@Component
+public class GrandpaService {
+    private final GrandpaState grandpaState;
+    private final BlockState blockState;
+
+    public GrandpaService(GrandpaState grandpaState, BlockState blockState) {
+        this.grandpaState = grandpaState;
+        this.blockState = blockState;
+    }
+
+    /**
+     * Finds and returns the block with the most votes in the GRANDPA prevote stage.
+     * If there are multiple blocks with the same number of votes, selects the block with the highest number.
+     * If no block meets the criteria, throws an exception indicating no valid GHOST candidate.
+     *
+     * @return GRANDPA GHOST block as a vote
+     */
+    public Vote getGrandpaGHOST() {
```

Comment:
```suggestion
    public Vote getGrandpaGhost() {
```

Just a personal pref, but I'd rather stick to the naming conventions unless absolutely necessary.